feature/preview/src/main/java/com/google/jetpackcamera/feature/preview/ui/PreviewScreenComponents.kt

```diff
@@ -0,0 +1,223 @@
+                    },
+                    onTap = { offset ->
+                        // tap to focus
+                        try {
+                            viewModel.tapToFocus(
+                                viewInfo.display,
+                                viewInfo.width,
+                                viewInfo.height,
+                                offset.x, offset.y
+                            )
+                            Log.d(TAG, "onTap $offset")
+                        } catch (e: UninitializedPropertyAccessException) {
+                            Log.d(TAG, "onTap $offset")
+                            e.printStackTrace()
+                        }
+                    }
+                )
+            },
+
+        contentAlignment = Alignment.Center
+    ) {
+        val maxAspectRatio: Float = maxWidth / maxHeight
+        val aspectRatio: Float =
+            previewUiState.currentCameraSettings.aspect_ratio.ratio.toFloat()
+        val shouldUseMaxWidth = maxAspectRatio <= aspectRatio
+        val width = if (shouldUseMaxWidth) maxWidth else maxHeight * aspectRatio
+        val height = if (!shouldUseMaxWidth) maxHeight else maxWidth / aspectRatio
+        Box(
+            modifier = Modifier
+                .width(width)
+                .height(height)
+                .transformable(state = transformableState)
+
+        ) {
+            CameraPreview(
+                modifier = Modifier
+                    .fillMaxSize(),
+                onSurfaceProviderReady = onSurfaceProviderReady,
+                onRequestBitmapReady = {
+                    val bitmap = it.invoke()
+                },
+                setSurfaceView = { s: View ->
+                    viewInfo = s
+                }
+            )
+        }
+    }
+}
+
+@Composable
+fun FlipCameraButton(
+    modifier: Modifier = Modifier,
+    enabledCondition: Boolean,
+    onClick: () -> Unit
+) {
+    Box(modifier = modifier) {
+        IconButton(
+            modifier = Modifier
+                .align(Alignment.Center)
+                .size(40.dp),
+            onClick = onClick,
+            enabled = enabledCondition
+        ) {
+            Icon(
+                imageVector = Icons.Filled.Refresh,
+                tint = Color.White,
+                contentDescription = stringResource(id = R.string.flip_camera_content_description),
+                modifier = Modifier.size(72.dp)
+            )
+        }
+    }
+}
+
+@Composable
+fun SettingsNavButton(modifier: Modifier, onNavigateToSettings: () -> Unit) {
+    IconButton(
+        modifier = modifier,
+        onClick = onNavigateToSettings,
+    ) {
+        Icon(
+            imageVector = Icons.Filled.Settings,
+            tint = Color.White,
+            contentDescription = stringResource(R.string.settings_content_description),
+            modifier = Modifier.size(72.dp)
+        )
+    }
+}
+
+@Composable
+fun ZoomScaleText(zoomScale: Float, show: Boolean) {
```

Comment:
Instead of passing show here, if show is false, lets skip adding this composable.